File: library-base/src/main/java/com/dashingqi/base/application/ApplicationController.java
package com.dashingqi.base.application;

import android.app.Application;
import android.util.Log;

import com.dashingqi.base.providers.application.IApplicationProvider;
import com.orhanobut.logger.Logger;

/**
 * @author : zhangqi
 * @time : 2020/5/20
 * desc :
 */
public class ApplicationController {

    public static boolean mIsDebug;

    public static void init(Application application, boolean isDebug) {
        mIsDebug = isDebug;
        for (IApplicationProvider provider : ApplicationManager.apps) {
            provider.init(application);
        }
    }

    public static void transformOnCreate() {
        Logger.d("transformOnCreate");
        for (IApplicationProvider provider : ApplicationManager.apps) {
            provider.onCreate();
        }
    }

    public static void transformOnLowMemory() {
        for (IApplicationProvider provider : ApplicationManager.apps) {
            provider.onLowMemory();
        }
    }
}

File: library-base/src/main/java/com/dashingqi/base/application/ApplicationController.kt
package com.dashingqi.base.application

import android.app.Application

/**
 * @author : zhangqi
 * @time : 2020/5/20
 * desc : Application的控制器
 * 主要用来
 *
 */
object ApplicationController {

    @JvmField
    var isDebug = true

    @JvmStatic
    fun init(application: Application, isDebug: Boolean) {
        this.isDebug = isDebug
        for (applicationProvider in ApplicationManager.apps) {
            applicationProvider.init(application)
        }
    }

    /**
     * 用于去执行各个实现了IApplicationProvider类的onCreate方法
     */
    @JvmStatic
    fun transformOnCreate() {
        for (applicationProvider in ApplicationManager.apps) {
            applicationProvider.onCreate()
        }
    }

    @JvmStatic
    fun transformOnLowMemory() {
        for (applicationProvider in ApplicationManager.apps) {
            applicationProvider.onLowMemory()
        }
    }
}